components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/SSOAgentConstants.java

```diff
@@ -70,6 +70,9 @@ private SSOAgentConstants() {
         public static final String OAUTH2_SAML2_GRANT_URL = "OAuth2SAML2GrantURL";
         public static final String SKIP_URIS = "SkipURIs";
         public static final String QUERY_PARAMS = "QueryParams";
+        public static final String SECRETKEY_FILEPATH =
+                "/components/org.wso2.carbon.identity.sso" +
+                        ".agent/src/main/java/org/wso2/carbon/identity/sso/agent/secretkey_temp.txt";
```

Comment:
I do not think this is correct. You are referring to a file path that is only available in our source. Once the binary jar is generated, this location will no longer apply.

You need to place the file inside your jar's META-INF/ folder. Then you can retrieve it via the API  `ClassLoader#getResourceAsStream("/META-INF/secretkey_temp.txt");`

Thanks.